ui/js/controllers/bugfiler.js

```diff
@@ -127,8 +127,13 @@ treeherder.controller('BugFilerCtrl', [
             summary = summary.split(" | ");
 
             for (var i=0; i < $scope.omittedLeads.length; i++) {
-                if (summary[0].search($scope.omittedLeads[i]) >= 0 && summary.length > 1) {
-                    summary.shift();
+                // In the case where only exact substring matches will provide usable bugs,
+                // Dropping things from omittedLeads will cause the substring to not match.
+                // Don't drop the lead in this case.
+                if ($scope.search_terms.length > 0 && !$scope.search_terms[0].includes($scope.omittedLeads[i])) {
```

Comment:
You could perhaps add to the comment something to the effect of, "If the omitted lead is in the search term, do not remove it from the summary.  If it's not in the search term, then DO remove it from the summary."

Also, we don't need two if statements here.  These should all just be ``&&``ed together.